larpandora/LArPandoraInterface/LArPandoraHelper.cxx

```diff
@@ -1500,6 +1503,17 @@ namespace lar_pandora {
     return larpandoraobj::PFParticleMetadata(pPfo->GetPropertiesMap());
   }
 
+  bool
+  LArPandoraHelper::IsDualPhase()
+  {
+    art::ServiceHandle<geo::Geometry const> theGeometry;
+    std::unordered_set<geo::_plane_sigtype> sigtypeSet;
+    for (readout::ROPID const& rID: theGeometry->IterateROPIDs()) {
+      sigtypeSet.insert(theGeometry->SignalType(rID));
+    }
+    return (!(sigtypeSet.count(geo::kInduction) && sigtypeSet.count(geo::kCollection)) && !sigtypeSet.count(geo::kMysteryType));
+  }
```

Comment:
Really excellent succinct logic assessment but my head does hurt a bit after parsing it.
I think @AndyChappell mentioned it but it might be better to favour clarity here.

One other thing to consider is what happens if a fourth type is added to the list of possible signal types.  To be honest, I'm not sure if we can be completely robust against all possible changes to dual phase signal type and changes to the possible signal types...
I don't have a good answer here...